File: IGListTest/IGListTest/Models/NSObject+IGListDiffable.m
//
//  NSObject+IGListDiffable.m
//  IGListTest
//
//  Created by liuruixuan on 2023/4/16.
//

#import "NSObject+IGListDiffable.h"

@implementation NSObject (IGListDiffable)

- (nonnull id<NSObject>)diffIdentifier {
    return self;
}

- (BOOL)isEqualToDiffableObject:(nullable id<IGListDiffable>)object {
    return [self isEqual:object];
}

@end

File: Marslink_Swift5/Marslink_Swift5/Models/NSObject+IGListDiffable.swift
//
//  NSObject+IGListDiffable.swift
//  Marslink_Swift5
//
//  Created by liuruixuan on 2023/4/17.
//

import Foundation
import IGListDiffKit

extension NSObject: ListDiffable {
    public func diffIdentifier() -> NSObjectProtocol {
        return self
    }

    public func isEqual(toDiffableObject object: ListDiffable?) -> Bool {
        return isEqual(object)
    }
}
